interview-questions.md

```diff
@@ -15,41 +15,45 @@ INSTRUCTOR EXAMPLE: What is a conditional statement?
 
 1. What is git? What is the difference between git and Github?
 
-  Your answer:
+  Your answer: Git is software on our computer for code while Github is the cloud where we can organize and keep our code.
 
-  Researched answer:
+  Researched answer: Git is a software tool on our local computer that allows us to track our projects over time.
+
+  Github is a hosting service for Git repositories.
 
 
 
 2. Which JavaScript operators will return a Boolean value?
 
-  Your answer:
+  Your answer: As learned in class, a Boolean value will return either true or false. We can use AND, OR, NOT to have a Boolean value returned.
 
-  Researched answer:
+  Researched answer: There are 6 different operators in JavaScript - Arithmetic, Comparison, Logical, Assignment, Conditional, and Ternary. Comparison operators include == , === , != , > , < , >= , <= and return true or false. Logical combines at least two conditions and can use && , || , !  - AND OR and NOT.
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer: JavaScript uses zero indexing meaning instead of beginning the count at 1 when counting the number of characters or values in a data type, it begins at 0. A value can be a number, string, or boolean value - true or false.
 
-  Researched answer:
+  Researched answer: According to w3resources, the index property is a read-only property containing the position of an expression.
+
+  There are different types of values. We've so far learned of primitive values in class - numbers, strings, boolean, and undefined.
 
 
 
 4. What is iteration?
 
-  Your answer:
+  Your answer: A single iteration would be completing 1 single cycle task that could be of many.
 
-  Researched answer:
+  Researched answer: We learned about for loops in class which can use iterations to process the code.
```

Comment:
This is a good example of a subject that should have been researched deeper. Iteration is core to computation and algorithms and there is a lot here to discover!